tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -106,488 +95,446 @@ describe(__filename, () => {
+    expect(
+      within(extensionCard).queryByText('Previous'),
+    ).not.toBeInTheDocument();
+    expect(within(extensionCard).queryByText('Next')).not.toBeInTheDocument();
   });
 
   it('renders AddonsByAuthorsCard for themes', () => {
     const userId = 123;
-    const { params, store } = signInUserWithUserId(userId);
+    const { params } = signInUserWithProps({ userId });
+    const user = getCurrentUser(store.getState().users);
 
-    const root = renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'addonType',
-      ADDON_TYPE_STATIC_THEME,
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'pageParam',
-      'page_t',
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp('paginate', true);
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'pathname',
-      `/user/${userId}/`,
-    );
+    expect(screen.getByText(`Themes by ${user.name}`)).toBeInTheDocument();
+    const themeCard = screen.queryAllByClassName('AddonsByAuthorsCard').item(1);
+    expect(screen.queryByText(themeCard, 'Previous')).not.toBeInTheDocument();
+    expect(screen.queryByText(themeCard, 'Next')).not.toBeInTheDocument();
   });
 
   it('renders a not found page if the API request is a 404', () => {
-    const { store } = dispatchSignInActions();
-    const errorHandler = new ErrorHandler({
-      id: 'some-error-handler-id',
-      dispatch: store.dispatch,
-    });
-    errorHandler.handle(
-      createApiError({
+    const userId = 123;
+    createFailedErrorHandler({
+      error: createApiError({
         response: { status: 404 },
         apiURL: 'https://some/api/endpoint',
         jsonResponse: { message: 'not found' },
       }),
-    );
+      id: createErrorHandlerId({ userId }),
+      store,
+    });
 
-    const root = renderUserProfile({ errorHandler, store });
+    renderUserProfile({ params: { userId } });
 
-    expect(root.find(NotFoundPage)).toHaveLength(1);
+    expect(
+      screen.getByText('Oops! We can’t find that page'),
+    ).toBeInTheDocument();
   });
 
   it('renders errors', () => {
-    const { store } = dispatchSignInActions();
-    const errorHandler = new ErrorHandler({
-      id: 'some-id',
-      dispatch: store.dispatch,
+    const userId = 123;
+    const errorString = 'unexpected error';
+    createFailedErrorHandler({
+      error: new Error(),
+      id: createErrorHandlerId({ userId }),
+      message: errorString,
+      store,
     });
-    errorHandler.handle(new Error('unexpected error'));
 
-    const root = renderUserProfile({ errorHandler, store });
+    renderUserProfile({ params: { userId } });
 
-    expect(root.find(ErrorList)).toHaveLength(1);
+    expect(screen.getAllByText(errorString)).not.toHaveLength(0);
   });
 
   it('renders an edit link', () => {
-    const root = renderUserProfile();
+    signInUserWithProps({ userId: 100 });
```

Comment:
If it's important that the user for whom the page is being rendered is also the signed in user, we should be explicitly rendering the page with that user, so either use `userId` of `100` both for `signInUserWithProps` and for  `renderUserProfile`, or capture the`userId` from `signInUserWithProps` and pass it into `renderUserProfile`. Whatever model you decide on, it's probably best to do it consistently across all of the tests.

I suppose if this is something that needs to be done often, you could have a `renderForSignedInUser`, which does both things and therefore ensures that the `userId`s are the same. If the rest of the test doesn't care about the `userId` (which this test, for example, seems not to), then you could just call this `renderForSignedInUser` helper and not worry about `userId` at all.

Reply:
I'm finding it difficult to be consistent while making so many changes in a single file 😞 

I've added a `signInUserAndRenderUserProfile` (naming?) helper for the common case of signing in and then immediately rendering the user profile for that user.